Documentation/gettingstarted/hubble.rst

```diff
@@ -109,6 +109,23 @@ Relay and Hubble's graphical UI.
                   --set hubble.relay.enabled=true \\
                   --set hubble.ui.enabled=true
 
+    .. group-tab:: Install Hubble into an existing Cilium deployment
+
+        .. include:: k8s-install-download-release.rst
+
+        Deploy Hubble with the following Helm command:
+
+            .. parsed-literal::
+
+               helm install cilium |CHART_RELEASE| \\
+                  --namespace kube-system \\
+                  --set agent=false \\
+                  --set operator.enabled=false \\
+                  --set hubble.enabled=true \\
+                  --set hubble.listenAddress=":4244" \\
+                  --set hubble.relay.enabled=true \\
+                  --set hubble.ui.enabled=true
```

Comment:
Actually, grepping for `hubble` in `install/kubernetes/cilium` suggests that this might not be enough, in particular to configure the relevant TLS settings in cilium-agent.

The idea behind this PR is to allow existing users to upgrade, ideally without interfering with other installers like kops. But maybe this is not sufficient for that use case.

Reply:
Yeah, mTLS here is probably the trickiest part. Because it requires additional secrets to be mounted on the Cilium pod if `hubble.listenAddress` is set. As long as we have `agent=false`, then the secrets will not be mounted to the existing `cilium` DaemonSet and therefore cannot be mounted by the Hubble server.

Enabling Hubble on top of an existing installation needs to basically do four things:

1. Generating the mTLS secrets for the communication between Hubble Relay and Cilium Agent. This is easy, it can be done either via Helm or `certgen` or simply manually with `kubectl create secret`.
2. Update the `cilium` ConfigMap to open the Hubble API via TCP (i.e. setting `hubble.listenAddress`). This probably needs to be a `kubectl patch` command as well, as the above command would overwrite the existing cilium ConfigMap.
3. Mount the mTLS secrets into the existing Cilium DaemonSet. This is harder and probably requires a `kubectl patch` command.
4. Deploy Hubble Relay and optionally UI as separate Deployments. This is easy can be done via Helm or a pre-generated YAML.

So in summary, we probably needs a `helm template` or `helm install` command very similar to the above one to create the mTLS secrets and the Relay/UI Deployment, followed by two `kubectl patch` commands to update the Cilium ConfigMap and DaemonSet.

Long-term, there are options to simplify this. A long hanging fruit is to unconditionally add the *optional* mTLS volume mount to the Cilium DaemonSet for the mTLS secrets. Because they are optional, they will not prevent the Cilium pod from running even if they don't exist yet.

Reply:
Thanks for the detailed outline @gandro . For now I think I'll put this on the shelf but if this becomes more common or if anyone finds this PR then we'd be happy to accept PRs to propose such a workflow :pray: